ludwig/trainers/trainer.py

```diff
@@ -225,6 +227,10 @@ def prepare(self):
             base_learning_rate *= lr_scale_fn(self.distributed.size())
         self.base_learning_rate = base_learning_rate
 
+        # Given that regex is supplied, freeze layers
```

Comment:
@ethanreidel Question: By placing this capability at the top-level "trainer.py" module, we imply that all subclasses will inherit this capability.  While for Computer Vision, based on yours and others' work, it will work and we believe will be a useful capability to have, what can we say about such trainers as `FineTuneTrainer` (for LLM fine-tuning) and `NoneTrainer` (for LLM predictions) -- and also the GBM Trainer (albeit not as popular nowadays).  In particular, I am curious whether or not it will work for LLM architectures and, if affirmative, what you think about how we can take advantage of it?  Thanks!

Reply:
@alexsherstinsky This wouldn't be particularly useful for the NoneTrainer, but it could be useful for the FineTuneTrainer. I left a comment above that might help simplify this by only enabling this parameter for the ECDTrainerConfig for now, which would then make it a no-op for LLMs and GBMs entirely.